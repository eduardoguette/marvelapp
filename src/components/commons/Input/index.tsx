import logoSearch from '@/assets/search.svg'
import styles from './input.module.css'
interface IInput extends React.InputHTMLAttributes<HTMLInputElement> {}
export const Input: React.FC<IInput> = (props) => {
  return (
    <label htmlFor={props.id} className={styles.label}>
      <img className={styles.icon} src={logoSearch} alt="sear" />
      <input {...props} className={styles.input} />
    </label>
  )
}
