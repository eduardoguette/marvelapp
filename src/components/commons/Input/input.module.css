.label {
  position: relative;
}
.icon {
  position: absolute;
  left: 0.5rem;
  top: 0;
  bottom: 0;
  margin: auto;
}
.input {
  border: none;
  border-bottom: 1px solid black;
  padding-inline: 2.5rem;
  padding-block: 1rem;
  width: 100%;
  text-transform: uppercase;

  &:focus {
    outline: none;
    border-bottom: 1px solid black;
  }
}
