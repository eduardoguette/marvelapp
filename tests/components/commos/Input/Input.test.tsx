import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'
import { Input } from '../../../../src/components/commons/Input/index'

describe('Input Component', () => {
  it('debería renderizar correctamente', () => {
    render(<Input id="search" placeholder="Buscar..." />)

    const inputElement = screen.getByPlaceholderText('Buscar...')
    expect(inputElement).not.toBeNull()
  })

  it('debería aplicar los props correctamente', () => {
    render(<Input id="search" placeholder="Buscar..." />)

    const inputElement = screen.getByPlaceholderText('Buscar...')
    expect(inputElement.getAttribute('id')).toBe('search')
    expect(inputElement.getAttribute('placeholder')).toBe('Buscar...')
  })

  it('debería permitir al usuario escribir en el input', () => {
    render(<Input id="search" placeholder="Buscar..." />)

    const inputElement = screen.getByPlaceholderText(
      'Buscar...'
    ) as HTMLInputElement
    fireEvent.change(inputElement, { target: { value: 'Test' } })
    expect(inputElement.value).toEqual('Test')
  })

  it('debería renderizar la imagen del icono de búsqueda correctamente', () => {
    render(<Input id="search" placeholder="Buscar..." />)

    const iconElement = screen.getByAltText('sear')
    expect(iconElement).not.toBeNull()
    expect(iconElement.getAttribute('src')).toBe('/src/assets/search.svg')
  })

  it('debería aplicar las clases CSS correctamente', () => {
    const { container } = render(<Input id="search" placeholder="Buscar..." />)

    expect(container).toMatchSnapshot()
    const inputElement = screen.getByPlaceholderText(
      'Buscar...'
    ) as HTMLInputElement
    expect(inputElement.getAttribute('placeholder')).toBe('Buscar...') 
    expect(inputElement.getAttribute('id')).toBe('search')
  })
})
