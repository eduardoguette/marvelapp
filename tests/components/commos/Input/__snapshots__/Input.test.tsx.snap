// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input Component > deber√≠a aplicar las clases CSS correctamente 1`] = `
<div>
  <label
    class="_label_97a894"
    for="search"
  >
    <img
      alt="sear"
      class="_icon_97a894"
      src="/src/assets/search.svg"
    />
    <input
      class="_input_97a894"
      id="search"
      placeholder="Buscar..."
    />
  </label>
</div>
`;
